### ISSUE TITLE
Compatibility

### ISSUE THREAD

**Issue by 0n0w1c** [Dummiez/Factorio-StableFoundations#4]
I am looking at trying to get my mod to be compatible. It uses the `surface.set_tiles` to place tiles under a building, after it is actually placed by the player. The result is a building on a foundation that does not receive the bonus. I tried adding 
`script.raise_event(defines.events.script_raised_built, {entity = entity, player_index = player.index})`
but that does not cause your mod to notice. A quick look at your code, but I do not see a handler for that event.

Would you be open to trying to incorporate that event and see if that helps? I do not know this will work, just kind of thinking it might.

https://mods.factorio.com/mod/Foundations

**Comment by Dummiez** [comment]
Will look into this

**Comment by 0n0w1c** [comment]
In looking more closely at the API, looks like on_script_raised_set_tiles would be more appropriate. I have tried fiddling with it, and so far, no luck. No errors, but seemingly not capturing the event. I will continue working on it and let you know if I find some reasonable measure of success.
Thanks for considering it, much appreciated.

**Comment by 0n0w1c** [comment]
script.on_event({ defines.events.on_player_built_tile, defines.events.on_robot_built_tile, defines.events.script_raised_set_tiles },

script.on_event({ defines.events.on_built_entity, defines.events.on_robot_built_entity, defines.events.script_raised_built },

Seems to work (with the most recent version of my mod) to capture the events. With the changes above, your mod does receive the built and then the tile built events, but no bonus is applied.

**Comment by 0n0w1c** [comment]
I should add that in order to get it to work, I needed to pass a parameter to surface.set_tiles to trigger the script_raised_set_tiles event. I was not able to capture the event if I tried raising the script_raised_set_tiles event.

**Comment by 0n0w1c** [comment]
The tileList is different: tilename,  position from script_raised_set_tiles. The other events have prototype, position. I messed around a bit and got it to recognize the event and even handle the different table structures... but I got weird results. Sometimes x4 bonus, then it moved up to x8 and then to x12 when repeatedly placing a building in the same location.
I could also remove all but the center tile from under a building and the bonus remained.
